### ISSUE TITLE
[Question] Join with grouping and custom ON

### ISSUE THREAD

**Issue by timothyvanderaerden** [typeorm/typeorm#1709]
**Issue type:**

 [x] question

**Database system/driver:**

 [x] `mysql` / `mariadb`

**TypeORM version:**

 [x] `0.2.0-alpha.13`

**Steps to reproduce or a small repository showing the problem:**

I looked through the issues and believe this is not a duplicate.

I want to create a query with an inner join where I group by a certain column and eventually use a custom ON. Something like this:

```sql
SELECT log.*
FROM log log
INNER JOIN (
    SELECT date, MAX(watchTime) AS watchTime
    FROM log
    GROUP BY date
) max
ON log.date = max.date AND log.watchTime = max.watchTime
WHERE log.identifier = 'someidentifier'
GROUP BY log.date, log.uniqueid
ORDER BY log.date DESC;
```
 So basically I just want the highest watch time for a certain date/time. I also could do something like this:

```sql
SELECT date, uniqueid, MAX(watchTime) as watchTime
FROM log
WHERE identifier = 'someidentifier'
GROUP BY date
ORDER BY date DESC;
```
And using TypeORM:
```ts
.createQueryBuilder('log')
        .select([
          'log.date',
          'MAX(log.watchTime) AS watchTime',
          ])
         ...
        .getRawMany();
```

But that would mean I get the Raw data and not the mapped entities. So is there a way to use the first MySQL query in TypeORM?

Thanks.

**Comment by pleerock** [comment]
There is a feature called `addSelectAndMap` that we are planning to implement in the near future.

**Comment by pleerock** [comment]
closing as it is a duplicate of #296
